### PR TITLE
fix(@angular/cli): no-op ng update --all

### LINKED ISSUE #12243: Expand documentation for `ng update`
I'm trying to do something that should be utterly trivial: update an Angular CLI 6.1.1 app to the latest version of 6.x.

Apparently, I'm supposed to use `ng update` to do this. But I have absolutely no idea what this command actually _does_, and the [docs](https://github.com/angular/angular-cli/wiki/update) don't help at all.

Consider:

```
>ng update
    We analyzed your package.json, there are some packages to update:

      Name                               Version                  Command to update
     --------------------------------------------------------------------------------
      @angular/cli                       6.1.1 -> 6.2.1           ng update @angular/cli
      @angular/core                      6.1.1 -> 6.1.7           ng update @angular/core
      rxjs                               6.2.2 -> 6.3.2           ng update rxjs


    There might be additional packages that are outdated.
    Or run ng update --all to try to update all at the same time.
```

Hmm...OK, I'm positive that more than cli and core need to be updated. Why aren't the other packages listed? There's a note that "there might be additional packages that are outdated". _Might?_ Can't the CLI see exactly what needs updating?

But OK, let's try it with `--all` and use `--dry-run` to try and see what will happen:

```
>ng update --all --dry-run
                  Package "bootstrap" has a missing peer dependency of "popper.js" @ "^1.14.3".
                  Package "@angular/compiler-cli" has an incompatible peer dependency to "typescript" (requires ">=2.7.2 <2.10", would install "3.0.3")
                  Package "@angular/compiler-cli" has an incompatible peer dependency to "typescript" (requires ">=2.7.2 <2.10", would install "3.0.3").
Incompatible peer dependencies found. See above.
```

OK, now I'm even more confused. Why is `ng update` doing anything with _bootstrap_? And `compiler-cli` wasn't even listed in the earlier list. Finally, if `ng update` is going to update `compiler-cli`, shouldn't it _know_ that `typescript` (obviously being a dependency of `compiler-cli`) would be getting updated too?

Basically, I have no idea what's going on or what's going to happen if I run `ng update --all`. What will be updated? Right now, it sounds like I just have to run it and hope that it knows what it's doing.

I think a lot of confusion could be avoided if the `ng update` docs were more than one (vague) sentence long. I'd take a shot at updating the docs myself, but I literally have no idea what this command actually does (other than "it updates something"). Could someone take a few minutes and expand the docs on this command? Because I assume that in-version updates like this are _far_ more common than full version updates.

### LINKED ISSUE #12261: ng update and ng update --all do not agree with each other
### Bug Report or Feature Request (mark with an `x`)
```
- [x ] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

```
Angular CLI: 6.1.2
Node: 8.11.4
OS: darwin x64
Angular: 
... 

Package                      Version
------------------------------------------------------
@angular-devkit/architect    0.7.2
@angular-devkit/core         0.7.2
@angular-devkit/schematics   0.7.2
@schematics/angular          0.7.2
@schematics/update           0.7.2
rxjs                         6.2.2
typescript                   2.7.2
```

### Repro steps

```
ng update
ng update --all
```

### The log given by the failure

```
ng update                  
    We analyzed your package.json and everything seems to be in order. Good work!

~/_git/vikings  master ✗                                                           17h22m ⚑  
▶ng update --all            
                  Package "@angular/compiler-cli" has an incompatible peer dependency to "typescript" (requires ">=2.7.2 <2.10", would install "3.0.3").
Incompatible peer dependencies found. See above.
```

### Desired functionality

If `ng update` says it's all clear, then the `ng update --all` should agree. Not sure which is right.

### LINKED ISSUE #13095: Package "@angular/compiler-cli" has an incompatible peer dependency to "typescript" (requires ">=3.1.1 <3.2", would install "3.2.1")
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

                  Node Version: v8.11.4
                  npm Version: 6.4.1
                  ng Version: 
                  Angular CLI: 7.0.4
                  Node: 8.11.4
                  OS: win32 x64
                  Angular: 7.0.2
                  ... animations, common, compiler, compiler-cli, core, forms
                  ... http, language-service, platform-browser
                  ... platform-browser-dynamic, router

                  Package                           Version
                  -----------------------------------------------------------
                  @angular-devkit/architect         0.10.4
                  @angular-devkit/build-angular     0.10.4
                  @angular-devkit/build-optimizer   0.10.4
                  @angular-devkit/build-webpack     0.10.4
                  @angular-devkit/core              7.0.4
                  @angular-devkit/schematics        7.0.4
                  @angular/cli                      7.0.4
                  @ngtools/webpack                  7.0.4
                  @schematics/angular               7.0.4
                  @schematics/update                0.10.4
                  rxjs                              6.3.3
                  typescript                        3.1.6
                  webpack                           4.19.1

### Repro steps
                  ng update --all

### The log given by the failure

                  Package "@angular/compiler-cli" has an incompatible peer dependency to "typescript" (requires ">=3.1.1 <3.2", would install "3.2.1")
                  Package "@angular/compiler-cli" has an incompatible peer dependency to "typescript" (requires ">=3.1.1 <3.2", would install "3.2.1").
Incompatible peer dependencies found. See above.

### LINKED ISSUE #15278: Suboptimal experience upgrading from 7 to 8
# 🐞 Bug report

### Command
`ng update`

### Is this a regression?
Not sure

### Description
I'm on Angular 7 and want all the goodness of Angular 8. I know that the CLI helps me do this, so I run `ng update` assuming this will help guide me to the latest version.

Here's what I experienced in the terminal:
```
^_^ ng update
Your global Angular CLI version (8.0.1) is greater than your local
version (7.3.8). The local Angular CLI version is used.

To disable this warning use "ng config -g cli.warnings.versionMismatch false".
    We analyzed your package.json, there are some packages to update:
    
      Name                               Version                  Command to update
     --------------------------------------------------------------------------------
      @angular/cdk                       7.3.7 -> 8.1.2           ng update @angular/cdk
      @angular/cli                       7.3.8 -> 8.2.1           ng update @angular/cli
      @angular/core                      7.2.12 -> 8.2.0          ng update @angular/core
      @angular/core                      7.2.12 -> 7.2.15         ng update @angular/core
      @angular/material                  7.3.7 -> 8.1.2           ng update @angular/material
      @ngrx/store                        7.4.0 -> 8.2.0           ng update @ngrx/store
      rxjs                               6.3.3 -> 6.5.2           ng update rxjs
    
    
    There might be additional packages that are outdated.
    Run "ng update --all" to try to update all at the same time.
    
^_^ ng update --all
Your global Angular CLI version (8.0.1) is greater than your local
version (7.3.8). The local Angular CLI version is used.

To disable this warning use "ng config -g cli.warnings.versionMismatch false".
                  Package "tsickle" has an incompatible peer dependency to "typescript" (requires "~3.4.1", would install "3.5.3")
                  Package "@angular/http" has an incompatible peer dependency to "@angular/core" (requires "7.2.15", would install "8.2.0")
                  Package "@angular/core" has an incompatible peer dependency to "zone.js" (requires "~0.9.1", would install "0.10.1")
                  Package "@angular/core" has an incompatible peer dependency to "zone.js" (requires "~0.9.1", would install "0.10.1").
                  Package "tsickle" has an incompatible peer dependency to "typescript" (requires "~3.4.1", would install "3.5.3").
Incompatible peer dependencies found. See above.

```

I talked with Vikram and now understand I need to run `ng update @angular/cli @angular/core`, which is also repeated on the site: https://angular.io/cli/update

However, I assume the CLI would have helped me figure this out in the terminal instead of giving me odd messages like `@angular/http` has an incompatible peer dependency to core.

It would have been nice for the CLI to understand my intentions and either perform the right commands for me, or guide me to the docs site or upgrade site to help me correct my commands.

## 🔬 Minimal Reproduction

You can check out my app and run it from there: https://github.com/crafted/crafted

## 🌍 Your Environment
<pre><code>
Angular CLI: 7.3.8
Node: 10.15.3
OS: linux x64
Angular: 7.2.12
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.13.8
@angular-devkit/build-angular      0.13.8
@angular-devkit/build-ng-packagr   0.13.8
@angular-devkit/build-optimizer    0.13.8
@angular-devkit/build-webpack      0.13.8
@angular-devkit/core               7.3.8
@angular-devkit/schematics         7.3.8
@angular/cdk                       7.3.7
@angular/cli                       7.3.8
@angular/fire                      5.1.2
@angular/material                  7.3.7
@ngtools/json-schema               1.1.0
@ngtools/webpack                   7.3.8
@schematics/angular                7.3.8
@schematics/update                 0.13.8
ng-packagr                         4.7.1
rxjs                               6.3.3
typescript                         3.2.4
webpack                            4.29.0
</code></pre>

### LINKED ISSUE #18813: Add the ability to skip dependencies with ng update
# 🚀 Feature request

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description
At present when I do `ng update --all`, the command fails when it finds incompatible dependencies(Attached image). I force update all dependencies and revert back incompatible dependencies through a shell script

Screen Shot 2020-09-17 at 4 46 35 PM

### Describe the solution you'd like
Provide the ability to skip dependencies through flag like below
```
ng update --all --skip "zone.js"
```

### Describe alternatives you've considered
I have a script that force updates the all dependencies and reverts incompatible dependencies 
```
### Update all dependencies
ng update --all --force

### Install compatible zone.js version
npm i zone.js@0.10.3 --save

```

### PR DESCRIPTION
**Pull request by alan-agius4**
`--all` functionality has been removed from `ng update` as updating multiple packages at once is not recommended. To update the depencencies in your workspace 'package.json' use the update command of your package manager.

Closes #15278
Closes #13095
Closes #12261
Closes #12243
Closes #18813

### CODE REVIEW

File: packages/angular/cli/commands/update-impl.ts [hunk 1]
```diff
@@ -307,6 +307,19 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
...
     }
 
+    if (options.all) {
+      const updateCmd = this.packageManager === PackageManager.Yarn
+        ? `'yarn upgrade-interactive' or 'yarn upgrade'`
+        : `'${this.packageManager} update'`;
+
+      this.logger.warn(`
+        '--all' functionality has been removed removed as updating multiple packages at once is not recommended.
+        To update the depencencies in your workspace 'package.json' use ${updateCmd} instead.
```
Comment (clydin) [inline packages/angular/cli/commands/update-impl.ts:317]:
```suggestion
        To update the non-Angular dependencies in your workspace 'package.json' use ${updateCmd} instead.
```

Should we mention to run the package manager update after using `ng update` for Angular dependencies?

Comment (alan-agius4) [reply packages/angular/cli/commands/update-impl.ts:317]: non-Angular dependencies might imply packages which are not under the `@angular` scope but still has migrations.

Maybe reword to `to update packages which don’t provide ng update capabilities`

Comment (clydin) [reply packages/angular/cli/commands/update-impl.ts:317]: sounds good

Comment (alan-agius4) [reply packages/angular/cli/commands/update-impl.ts:317]: Yeah we should. I’ll think of a way to word that

Comment (alan-agius4) [reply packages/angular/cli/commands/update-impl.ts]: PTAL

File: packages/angular/cli/commands/update-impl.ts [hunk 2]
```diff
@@ -307,6 +307,20 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
       );
     }
 
+    if (options.all) {
+      const updateCmd = this.packageManager === PackageManager.Yarn
+        ? `'yarn upgrade-interactive' or 'yarn upgrade'`
+        : `'${this.packageManager} update'`;
+
+      this.logger.warn(`
+        '--all' functionality has been removed removed as updating multiple packages at once is not recommended.
```
Comment (clydin) [inline packages/angular/cli/commands/update-impl.ts:316]:
```suggestion
        '--all' functionality has been removed as updating multiple packages at once is not recommended.
```

**Review by clydin — APPROVED** [review]

**Pull request merged**